examples/plot_fpca_inverse_transform_outl_detection.py

```diff
@@ -0,0 +1,231 @@
+)
+plt.title('train and test samples')
+fig.show()
+
+#####################################################################
+# We fit an FPCA with an arbitrary low number of components
+# compared to the input dimension (grid size).
+# We compute the relative RE
+# of both the training and test samples, and plot the pdf estimates.
+# Errors are normalized w.r.t L2-norms of each sample
+# to remove (explained) variance from the scale error.
+
+q = 5
+fpca_clean = FPCA(n_components=q)
+fpca_clean.fit(train_set)
+train_set_hat = fpca_clean.inverse_transform(
+    fpca_clean.transform(train_set)
+)
+
+err_train = lp_distance(
+    train_set,
+    train_set_hat,
+    p=2
+) / lp_norm(train_set, p=2)
+
+test_set_hat = fpca_clean.inverse_transform(
+    fpca_clean.transform(test_set)
+)
+err_test = lp_distance(
+    test_set,
+    test_set_hat,
+    p=2
+) / lp_norm(test_set, p=2)
+
+###########################################################################
+# We plot the density of the REs,
+# both unconditionnaly (grey and blue) and conditionnaly (orange and red),
+# to the rule if error >= threshold then it is an outlier.
+# The threshold is computed from RE of the training samples as
+# the quantile of probability 0.99.
+# In ohter words, a sample whose RE is higher than the threshold is unlikely
+# approximated as a training sample with (low) probability 0.01.
+x_density = np.linspace(0., 1.6, num=10**3)
+density_train_err = gaussian_kde(err_train)
+density_test_err = gaussian_kde(err_test)
+err_thresh = np.quantile(err_train, 0.99)
+
+density_test_err_outl = gaussian_kde(err_test[err_test >= err_thresh])
+density_test_err_inli = gaussian_kde(err_test[err_test < err_thresh])
+
+# density estimate of train errors
+plt.plot(x_density, density_train_err(x_density),
+         label='Error train', color='grey')
+
+# density estimate of test errors
+plt.plot(x_density, density_test_err(x_density),
+         label='Error test (outliers+nonoutliers', color='C0')
+
+# outlyingness threshold
+plt.vlines(err_thresh,
+           ymax=max(density_train_err(x_density)), ymin=0.,
+           label='thresh=quantile(p=0.99)',
+           linestyles='dashed', color='black')
+
+# density estimate of the error of test samples flagged as outliers
+plt.plot(x_density, density_test_err_outl(x_density),
+         label='Error test>= thresh (outliers)', color='C1')
+
+# density estimate of the error of test samples flagged as nonoutliers
+plt.plot(x_density, density_test_err_inli(x_density),
+         label='Error test< thresh (nonoutliers)', color='C3')
+
+plt.xlabel('Relative L2-norm reconstruction errors')
+plt.ylabel('Density (unormalized)')
+plt.title('Densities of reconstruction errors with {} components'.format(q))
+plt.legend()
+plt.show()
+
+####################################################################
+# We can check that the outliers are all detected with this method,
+# with no false positive (wrongly) in the test set.
+print('Flagged outliers: \n',
```

Comment:
I think I prefer to call print twice than to pass two arguments and embed a newline.